R/utils.R

```diff
@@ -39,14 +25,11 @@ csv_reader <- function(file, verbose = FALSE, guess_max = 1000, ...) {
   }
   if (verbose) {
     message("Downloading data from ", file)
-    data <- read_csv_fun(file, ..., guess_max = guess_max)
+    data <- read_csv_fun(file, progress = TRUE, ..., guess_max = guess_max)
```

Comment:
I think the reason this was coded this way is that there was a need for more than one type of csv reading function as `vroom` was so badly behaved. This appears to have got more stable so perhaps we can drop this notional support and so be able to make the changes you have suggested here. What do you think @RichardMN?

Reply:
I've just run quick tests on two of the countries which used to be problematic for `vroom` and they are no longer reporting problems. This would mean we can probably remove some of this work-around, and I may need to go and clean the Belgium code of some of its old fixes.

``` r
library(vroom)
library(covidregionaldata)

be <- Belgium$new()
be_data <- be$download()
#> Downloading data from https://epistat.sciensano.be/Data/COVID19BE_MORT.csv
#> Rows: 7277 Columns: 5
#> ── Column specification ────────────────────────────────────────────────────────
#> Delimiter: ","
#> chr  (3): REGION, AGEGROUP, SEX
#> dbl  (1): DEATHS
#> date (1): DATE
#> 
#> ℹ Use `spec()` to retrieve the full column specification for this data.
#> ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#> Downloading data from https://epistat.sciensano.be/Data/COVID19BE_CASES_AGESEX.csv
#> 
#> Rows: 120513 Columns: 6
#> ── Column specification ────────────────────────────────────────────────────────
#> Delimiter: ","
#> chr  (4): PROVINCE, REGION, AGEGROUP, SEX
#> dbl  (1): CASES
#> date (1): DATE
#> 
#> ℹ Use `spec()` to retrieve the full column specification for this data.
#> ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#> Downloading data from https://epistat.sciensano.be/Data/COVID19BE_HOSP.csv
#> 
#> Rows: 6897 Columns: 10
#> ── Column specification ────────────────────────────────────────────────────────
#> Delimiter: ","
#> chr  (2): PROVINCE, REGION
#> dbl  (7): NR_REPORTING, TOTAL_IN, TOTAL_IN_ICU, TOTAL_IN_RESP, TOTAL_IN_ECMO...
#> date (1): DATE
#> 
#> ℹ Use `spec()` to retrieve the full column specification for this data.
#> ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
problems(be$data$raw$deaths)
#> # A tibble: 0 × 5
#> # … with 5 variables: row <int>, col <int>, expected <chr>, actual <chr>,
#> #   file <chr>
problems(be$data$raw$hosp)
#> # A tibble: 0 × 5
#> # … with 5 variables: row <int>, col <int>, expected <chr>, actual <chr>,
#> #   file <chr>
problems(be$data$raw$main)
#> # A tibble: 0 × 5
#> # … with 5 variables: row <int>, col <int>, expected <chr>, actual <chr>,
#> #   file <chr>

it <- Italy$new()
it_data <- it$download()
#> Downloading data from https://raw.githubusercontent.com/pcm-dpc/COVID-19/master/dati-regioni/dpc-covid19-ita-regioni.csv
#> Rows: 13608 Columns: 30── Column specification ────────────────────────────────────────────────────────
#> Delimiter: ","
#> chr   (8): stato, codice_regione, denominazione_regione, note, note_test, no...
#> dbl  (21): lat, long, ricoverati_con_sintomi, terapia_intensiva, totale_ospe...
#> dttm  (1): data
#> ℹ Use `spec()` to retrieve the full column specification for this data.
#> ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
problems(it$data$raw$main)
#> # A tibble: 0 × 5
#> # … with 5 variables: row <int>, col <int>, expected <chr>, actual <chr>,
#> #   file <chr>
```

<sup>Created on 2021-12-02 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
If we go with Hugo's PR as is that won't cause any issues will it and then can clean up code later?